witness/ethereum/README.md

```diff
@@ -0,0 +1,52 @@
+# Ethereum Witness
+
+## To run tests
+
+### Install `pipenv` and `npm`
+
+```bash
+sudo apt install pipenv nodejs npm
+```
+
+### Install `ganache-cli`
+
+- Instructions from [here](https://github.com/trufflesuite/ganache-cli); run
+  the following command:
+
+```bash
+npm install --global ganache-cli
+```
+
+### Install `eth-brownie`
+
+- Since our project uses `pipenv` to manage its python environment, we can
+  just run the following:
+
+```bash
+pipenv install
+```
+
+(run from this directory)
+
+### Finally, run the tests:
+
+```bash
+pipenv run brownie test
+```
+
+## To run tests using docker
+
+- cd into this directory
+- `docker run --rm -it $(docker build -q .)`
+  - This builds the container, runs it, and removes it after the run is
+    complete.
+
+## Explaination
+
+The Ethereum Witness contracts act as an on-chain registry of merkle tree
+log states that are otherwise maintained off-chain. The contract handles
+updates to logs and helps ensure that log operators cannot provide a split
+view of the log prior to a confirmed checkpoint.
+
+Notably, the contract verifies consistency proofs of new merkle roots it
```

Comment:
Small thought: I think the higher-level readme we're planning is the right place to elaborate on the differences between using "normal" consistency proofs and using compact ranges, but it might be worth adding a quick note here saying that `RootWitness` and `CompactRangeWitness` are separate contracts that verify consistency in two different ways?  We can also wait to add this until we have the higher-level readme in place if that makes more sense.

Reply:
I updated this section; please take a look and let me know what you think!

Reply:
Looks good to me!